File: frontend/src/shared/models/database/discord-bot-dto.ts
export class DiscordBotDto {
  id: number = 0;

  name: string = "";
  webhookUrl: string = "";
  botName: string = "";
  avatarUrl: string = "";
  rateLimitSeconds: number = 10;
  createdOn?: string;
  updatedOn?: string | null;
  flowStepsCount: number = 0;

  constructor(data: Partial<DiscordBotDto> = {}) {
    Object.assign(this, { ...data });
  }
}

export const RATE_LIMIT_MIN_SECONDS = 2;
export const RATE_LIMIT_MAX_SECONDS = 300;

File: backend/Core/Models/Dtos/Database/DiscordBotDto.cs
namespace Core.Models.Dtos
{
    public class DiscordBotDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string WebhookUrl { get; set; } = string.Empty;
        public string BotName { get; set; } = string.Empty;
        public string AvatarUrl { get; set; } = string.Empty;
        public int RateLimitSeconds { get; set; } = 10;

        public DateTime CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }

        /// <summary>Steps wired to this bot. Deleting one that is in use is refused.</summary>
        public int FlowStepsCount { get; set; }
    }

    /// <summary>A one-off send straight from the settings form, before anything is saved.</summary>
    public class TestDiscordBotDto
    {
        public string WebhookUrl { get; set; } = string.Empty;
        public string BotName { get; set; } = string.Empty;
        public string AvatarUrl { get; set; } = string.Empty;
    }
}
